datafusion/functions/src/strings.rs

```diff
@@ -18,102 +18,12 @@
 use std::mem::size_of;
 
 use arrow::array::{
-    make_view, Array, ArrayAccessor, ArrayDataBuilder, ArrayIter, ByteView,
-    GenericStringArray, LargeStringArray, OffsetSizeTrait, StringArray, StringViewArray,
-    StringViewBuilder,
+    make_view, Array, ArrayAccessor, ArrayDataBuilder, ByteView, LargeStringArray,
+    StringArray, StringViewArray, StringViewBuilder,
 };
 use arrow::datatypes::DataType;
 use arrow_buffer::{MutableBuffer, NullBuffer, NullBufferBuilder};
 
-/// Abstracts iteration over different types of string arrays.
```

Comment:
I double checked and this was made public in arrow in this PR:
- https://github.com/apache/arrow-rs/pull/6720/files#r1844543809

Thanks @tlm365 (for both changes!)